roscala/src/main/scala/coop/rchain/rosette/VirtualMachine.scala

```diff
@@ -779,7 +779,7 @@ object VirtualMachine {
 
   def execute(op: OpXferRsltToDest, state: VMState): VMState =
     state
-      .set(_ >> 'loc)(LocationAtom(state.code.lit(op.lit)))
+    //.set(_ >> 'loc)(LocationAtom(state.code.lit(op.lit))) //TODO probably we should change `atom` field only
```

Comment:
Also for another PR but I leave it here so you know: `Location` as a sealed trait does not make sense. Even more I think we can get rid of this line and the `atom` part of `Location` completely. This is because in Rosette I only see `atom` being used as a store for `ExpandedLocation` and as far as I can see the only use of `ExpandedLocation` is to have a printable version of the encoded `locfields`.

Reply:
@KentShikama do you agree with me on `Location.atom` only be used to store an `ExpandedLocation`?

Reply:
I think the main purpose of Location.atom is to figure out whether two locations are equal (see https://github.com/rchain/rchain/blob/dev/rosette/src/Location.cc#L30). 

For example, LocLimbo is identified by its atom.

(gdb) p LocLimbo.atom
$4 = (Ob *) 0x157

If any other Location object has 0x157 as its atom, then it is a LocLimbo.

The atom is updated directly by writing into the first bits of an Location object using macros. For example, the following calls all affect the atom of the Location.

SET_LEXVAR_IND(loc, (indirect != 0));
SET_LEXVAR_LEVEL(loc, level);
SET_LEXVAR_OFFSET(loc, offset);
SET_LEXVAR_TYPE(loc, LT_LexVariable);
SET_GENERIC_TAG(loc, OTlocation);

(from https://github.com/rchain/rchain/blob/dev/rosette/src/Location.cc#L341-L345).

I haven't really had a look at the Scala reimplementation but if you can compare Locations without atoms then I think you can get rid of the atom indeed.

Reply:
Ignore the above comment.

During a discussion with @tymm , we realized that Location is a union not a struct

union Location {
    Ob* atom;
    unsigned int locfields;
};

So a clearer way to put it is that the atom is the Location.